python/TestHarness/testers/RunApp.py

```diff
@@ -254,16 +254,16 @@ def testExitCodes(self, moose_dir, options, output):
             elif self.exit_code != 0 and specs['should_crash'] == False:
                 # Let's look at the error code to see if we can perhaps further split this out later with a post exam
                 reason = 'CRASH'
-                return "\n\nExit Code: " + str(self.exit_code)
```

Comment:
This is what was causing TestHarness unittests to pass. We were exiting this function before establishing a status for the test.